roop/core.py

```diff
@@ -209,6 +209,7 @@ def start(preview_callback = None):
         process_video_multi_cores(args.source_img, args.frame_paths)
     else:
         process_video(args.source_img, args.frame_paths)
+    torch.cuda.empty_cache() 
```

Comment:
You need to wrap this in a condition for nvidia otherwise it gonna kill amd platforms

Reply:
pytorch doc says it should work on amd platform: https://pytorch.org/docs/stable/notes/hip.html#memory-management

Reply:
@s0md3v Told me a different story

Reply:
i added a gpu check (both amd and nvidia)

Reply:
since i don't have amd card, idk why

i changed the file properly